airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/avro/JsonToAvroSchemaConverter.java

```diff
@@ -254,14 +265,6 @@ Schema getNullableFieldTypes(final String fieldName, final JsonNode fieldDefinit
       if (!nonNullFieldTypes.contains(NULL_SCHEMA)) {
         nonNullFieldTypes.add(0, NULL_SCHEMA);
       }
-      // Logical types are converted to a union of logical type itself and string. The purpose is to
-      // default the logical type field to a string, if the value of the logical type field is invalid and
-      // cannot be properly processed.
-      if ((nonNullFieldTypes
-          .stream().anyMatch(schema -> schema.getLogicalType() != null)) &&
-          (!nonNullFieldTypes.contains(STRING_SCHEMA))) {
-        nonNullFieldTypes.add(STRING_SCHEMA);
-      }
```

Comment:
Wait, you cannot remove this code block. This is necessary to handle logical types comprehensively.

At this point, I think it is too messy to modify this converter to make it work for both Avro and BigQuery. Can you just duplicate the code, and create one just for BigQuery?